client/src/components/EmailAlertSignup.tsx

```diff
@@ -64,20 +73,52 @@ const BuildingSubscribeWithoutI18n = (props: BuildingSubscribeProps) => {
   return (
     <I18n>
       {({ i18n }) => (
-        <div className="table-content building-subscribe">
-          {!(subscriptions && !!subscriptions?.find((s) => s.bbl === bbl)) ? (
-            <button
-              className="button is-primary"
-              onClick={() => subscribe(bbl, housenumber, streetname, zip, boro)}
-            >
-              <Trans>Get updates</Trans>
-            </button>
-          ) : verified ? (
-            showSubscribed()
-          ) : (
-            showEmailVerification(i18n)
-          )}
-        </div>
+        <>
+          <div className="table-content building-subscribe">
+            {!(subscriptions && !!subscriptions?.find((s) => s.bbl === bbl)) ? (
+              <button
+                className="button is-primary"
+                onClick={() =>
+                  subscriptions.length <= SUBCSCRIPTION_LIMIT
+                    ? subscribe(bbl, housenumber, streetname, zip, boro)
+                    : setShowSubscriptionLimitModal(true)
+                }
+              >
+                <Trans>Get updates</Trans>
+              </button>
+            ) : verified ? (
+              showSubscribed()
+            ) : (
+              showEmailVerification(i18n)
+            )}
+          </div>
+
+          <Modal
+            key={1}
+            showModal={showSubscriptionLimitModal}
+            width={40}
+            onClose={() => setShowSubscriptionLimitModal(false)}
+          >
+            <Trans render="h4">You have reached the maximum number of building updates</Trans>
+            <Trans>
+              At this time we can only support {SUBCSCRIPTION_LIMIT} buildings in each email. Please
+              visit your{" "}
+              <Link to={isLegacyPath(pathname) ? legacy.account.settings : account.settings}>
```

Comment:
are we supporting the account architecture in the legacy version? If we haven't had this conversation yet with the team, it might be worth discussing in an upcoming meeting?

Reply:
you're right, we aren't including legacy - just confirmed with Joel.